utf8.h

```diff
@@ -591,6 +590,11 @@ void *utf8ncpy(void *utf8_restrict dst, const void *utf8_restrict src,
     }
   }
 
+  for ( check_index = index - 1; check_index > 0 && 0x80 == (0xc0 & d[check_index]); check_index--);
```

Comment:
Can you remove the single line for? I always prefer having { } even if they are empty (and put a new line between the for and if please!.

Minor thing too - I think this line would fail with a clang-format run, could you run git clang-format -f? Otherwise I can do it just as I merge.

Reply:
Updated. `git clang-format -f` didn't find any issues, but I changed the for loop for your preference.